nevermined_metadata/app/assets.py

```diff
@@ -65,28 +67,92 @@ def get_metadata(did):
     try:
         asset_record = get_dao().get(did)
         metadata = _get_metadata(asset_record['service'])
-        return Response(_sanitize_record(metadata), 200, content_type='application/json')
+        return Response(
+            _sanitize_record(metadata), 200, content_type='application/json'
+        )
     except Exception as e:
         logger.error(e)
         return f'{did} asset DID is not in MetadataDB', 404
 
 
+@assets.route('/agreement/<agreementId>', methods=['GET'])
+def get_agreement(agreementId):
+    """Get the ServiceAgreement with a particular agreementID
+    swagger_from_file: docs/get_agreement.yml
+    """
+    try:
+        asset_record = get_dao().get(agreementId)
+        return Response(
+            _sanitize_record(asset_record), 200, content_type='application/json'
+        )
+    except Exception as e:
+        logger.error(e)
+        return f'{agreementId} agreement is not in the MetadataDB', 404
+
+
+@assets.route('/agreement', methods=['POST'])
+def register_agreemet():
+    """Register a ServiceAgreement.
+    swagger_from_file: docs/register_agreement.yml
+    """
+    assert isinstance(request.json, dict), 'invalid payload format.'
+    required_attributes = [
+        'type',
+        'index',
+        'serviceEndpoint',
+        'templateId',
+        'attributes',
+    ]
+    data = request.json
+    if not data:
+        logger.error(f'request body seems empty.')
+        return 400
+    msg, status = check_required_attributes(required_attributes, data, 'register')
+    if msg:
+        return msg, status
+    # TODO: are these checks necessary?
```

Comment:
@r-marques Which of these checks are needed?

Reply:
only the dates